plugins/common/opcua/input/input_client.go

```diff
@@ -368,19 +374,111 @@ func (o *OpcUAInputClient) InitNodeMetricMapping() error {
 	return nil
 }
 
-func (o *OpcUAInputClient) InitNodeIDs() error {
+func (o *OpcUAInputClient) InitNodeIDs(ctx context.Context) error {
 	o.NodeIDs = make([]*ua.NodeID, 0, len(o.NodeMetricMapping))
 	for _, node := range o.NodeMetricMapping {
-		nid, err := ua.ParseNodeID(node.Tag.NodeID())
-		if err != nil {
-			return err
+		var nid *ua.NodeID = nil
+
+		if node.Tag.Browse && node.Tag.IdentifierType == "s" {
+			browsedNodeID, err := o.browseNode(ctx, node.Tag.Namespace, node.Tag.Identifier)
+			if err != nil {
+				o.Log.Errorf("Error browsing node %s: %s", node.Tag.FieldName, err)
+				// continue, other nodes may be browseable
+				// also, add the node as a string identifier. The server will not be able to return data
+				// but we must add a node to the o.NodeIDs array in order for it to match the o.NodeMetricMappings
+				parsedNodeID, err := ua.ParseNodeID(node.Tag.NodeID())
+				if err != nil {
+					return err
+				}
+				nid = parsedNodeID
+			} else {
+				nid = browsedNodeID
+			}
+		} else {
+			parsedNodeID, err := ua.ParseNodeID(node.Tag.NodeID())
+			if err != nil {
+				return err
+			}
+			nid = parsedNodeID
 		}
+
 		o.NodeIDs = append(o.NodeIDs, nid)
 	}
 
 	return nil
 }
 
+func parseNamespaceId(ns string) (uint16, error) {
+	nsId, err := strconv.ParseUint(ns, 10, 16)
+	if err != nil {
+		return 0, fmt.Errorf("invalid namespace number: %s", err)
+	}
+
+	return uint16(nsId), nil
+}
```

Comment:
Please merge this into the respective calling places as this is not worth a wrapper.